### PR TITLE
Rebasing ocs-ci 4.17 Changed the required patch files

### PR DESCRIPTION
**Pull request by Shilpi-Das1**
**Output setup script before applying patch**
```
Generating consolidated patch file /root/shilpi-rebase/ocs-ci.patch from /root/shilpi-rebase/ocs-upi-kvm/files/ocs-ci/
checking file ocs_ci/ocs/benchmark_operator.py
checking file ocs_ci/ocs/amq.py
checking file ocs_ci/templates/workloads/amq/benchmark/values.yaml
checking file ocs_ci/templates/workloads/amq/hello-world-consumer.yaml
checking file ocs_ci/templates/workloads/amq/hello-world-producer.yaml
checking file tests/cross_functional/scale/test_scale_osds_fill_75%_reboot_workers.py
checking file tests/functional/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py
Hunk #2 succeeded at 48 (offset 2 lines).
checking file tests/functional/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py
checking file tests/functional/z_cluster/cluster_expansion/test_delete_pod.py
checking file tests/functional/z_cluster/cluster_expansion/test_node_expansion.py
checking file tests/functional/z_cluster/cluster_expansion/test_verify_ceph_csidriver_runs_on_non_ocs_nodes.py
checking file tests/functional/z_cluster/cluster_expansion/test_crashcollector_pod_existence_on_ceph_pods_running_nodes.py
checking file tests/functional/z_cluster/nodes/test_node_replacement_proactive.py
checking file tests/cross_functional/scale/test_pv_scale_and_respin_ceph_pods.py
checking file ocs_ci/ocs/node.py
Hunk #1 succeeded at 264 (offset 1 line).
checking file ocs_ci/deployment/deployment.py
Hunk #1 succeeded at 2517 (offset 247 lines).
checking file ocs_ci/templates/workloads/amq/kafkadrop.yaml
checking file ocs_ci/ocs/resources/storage_cluster.py
Hunk #1 succeeded at 556 (offset 1 line).
checking file ocs_ci/ocs/resources/pod.py
Hunk #1 succeeded at 2091 (offset 10 lines).
checking file ocs_ci/templates/workloads/logwriter/cephfs.logreader.yaml
checking file ocs_ci/templates/workloads/logwriter/cephfs.logwriter.yaml
checking file ocs_ci/templates/workloads/logwriter/cephfs.reproducer.yaml
checking file ocs_ci/deployment/helpers/lso_helpers.py
checking file tests/functional/z_cluster/test_ceph_default_values_check.py
checking file tests/functional/z_cluster/test_must_gather.py
checking file tests/functional/object/mcg/test_s3_with_java_sdk.py
checking file tests/functional/object/mcg/test_noobaa_secret.py
checking file tests/functional/z_cluster/test_hugepages.py
checking file tests/functional/z_cluster/test_rook_ceph_operator_log_type.py
checking file tests/functional/monitoring/prometheus/alerts/test_noobaa.py
checking file tests/functional/object/mcg/test_mcg_resources_disruptions.py
checking file tests/functional/object/mcg/test_host_node_failure.py
checking file tests/functional/z_cluster/nodes/test_nodes_restart.py
Hunk #2 succeeded at 140 (offset 1 line).
checking file ocs_ci/helpers/helpers.py
Hunk #1 succeeded at 99 (offset -1 lines).
checking file ocs_ci/utility/deployment.py
Hunk #1 succeeded at 138 (offset 5 lines).
checking file tests/functional/z_cluster/test_rook_ceph_log_rotate.py
checking file ocs_ci/templates/CSI/cephfs/pod.yaml
checking file ocs_ci/templates/CSI/rbd/pod.yaml
checking file ocs_ci/templates/app-pods/nginx.yaml
checking file ocs_ci/templates/app-pods/raw_block_pod.yaml
Patching ocs-ci...
patching file ocs_ci/ocs/benchmark_operator.py
patching file ocs_ci/ocs/amq.py
patching file ocs_ci/templates/workloads/amq/benchmark/values.yaml
patching file ocs_ci/templates/workloads/amq/hello-world-consumer.yaml
patching file ocs_ci/templates/workloads/amq/hello-world-producer.yaml
patching file tests/cross_functional/scale/test_scale_osds_fill_75%_reboot_workers.py
patching file tests/functional/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py
Hunk #2 succeeded at 48 (offset 2 lines).
patching file tests/functional/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py
patching file tests/functional/z_cluster/cluster_expansion/test_delete_pod.py
patching file tests/functional/z_cluster/cluster_expansion/test_node_expansion.py
patching file tests/functional/z_cluster/cluster_expansion/test_verify_ceph_csidriver_runs_on_non_ocs_nodes.py
patching file tests/functional/z_cluster/cluster_expansion/test_crashcollector_pod_existence_on_ceph_pods_running_nodes.py
patching file tests/functional/z_cluster/nodes/test_node_replacement_proactive.py
patching file tests/cross_functional/scale/test_pv_scale_and_respin_ceph_pods.py
patching file ocs_ci/ocs/node.py
Hunk #1 succeeded at 264 (offset 1 line).
patching file ocs_ci/deployment/deployment.py
Hunk #1 succeeded at 2517 (offset 247 lines).
patching file ocs_ci/templates/workloads/amq/kafkadrop.yaml
patching file ocs_ci/ocs/resources/storage_cluster.py
Hunk #1 succeeded at 556 (offset 1 line).
patching file ocs_ci/ocs/resources/pod.py
Hunk #1 succeeded at 2091 (offset 10 lines).
patching file ocs_ci/templates/workloads/logwriter/cephfs.logreader.yaml
patching file ocs_ci/templates/workloads/logwriter/cephfs.logwriter.yaml
patching file ocs_ci/templates/workloads/logwriter/cephfs.reproducer.yaml
patching file ocs_ci/deployment/helpers/lso_helpers.py
patching file tests/functional/z_cluster/test_ceph_default_values_check.py
patching file tests/functional/z_cluster/test_must_gather.py
patching file tests/functional/object/mcg/test_s3_with_java_sdk.py
patching file tests/functional/object/mcg/test_noobaa_secret.py
patching file tests/functional/z_cluster/test_hugepages.py
patching file tests/functional/z_cluster/test_rook_ceph_operator_log_type.py
patching file tests/functional/monitoring/prometheus/alerts/test_noobaa.py
patching file tests/functional/object/mcg/test_mcg_resources_disruptions.py
patching file tests/functional/object/mcg/test_host_node_failure.py
patching file tests/functional/z_cluster/nodes/test_nodes_restart.py
Hunk #2 succeeded at 140 (offset 1 line).
patching file ocs_ci/helpers/helpers.py
Hunk #1 succeeded at 99 (offset -1 lines).
patching file ocs_ci/utility/deployment.py
Hunk #1 succeeded at 138 (offset 5 lines).
patching file tests/functional/z_cluster/test_rook_ceph_log_rotate.py
patching file ocs_ci/templates/CSI/cephfs/pod.yaml
patching file ocs_ci/templates/CSI/rbd/pod.yaml
patching file ocs_ci/templates/app-pods/nginx.yaml
patching file ocs_ci/templates/app-pods/raw_block_pod.yaml
Requirement already satisfied: pip in /root/shilpi-rebase/venv/lib/python3.9/site-packages (21.2.3)
Collecting pip
  Using cached pip-24.1.2-py3-none-any.whl (1.8 MB)
Collecting setuptools==63.2.0
```
**Output setup script after applying the changed patch files**
```
~/shilpi-rebase/ocs-upi-kvm/src/ocs-ci ~/shilpi-rebase/ocs-upi-kvm/scripts
patchfiles=../../files/ocs-ci/ocs-ci-00-ripsaw.patch ../../files/ocs-ci/ocs-ci-01-strimzi.patch ../../files/ocs-ci/ocs-ci-02-skipscaletest.patch ../../files/ocs-ci/ocs-ci-03-addcapacity-skip.patch ../../files/ocs-ci/ocs-ci-04-skip-memoryleak.patch ../../files/ocs-ci/ocs-ci-05-drain.patch ../../files/ocs-ci/ocs-ci-06-check-sc-exists.patch ../../files/ocs-ci/ocs-ci-07-kafka-drop.patch ../../files/ocs-ci/ocs-ci-08-nodename-validation.patch ../../files/ocs-ci/ocs-ci-09-podname-change.patch ../../files/ocs-ci/ocs-ci-10-logwriter.patch ../../files/ocs-ci/ocs-ci-11-lso-pod-security.patch ../../files/ocs-ci/ocs-ci-12-skip-known-failures.patch ../../files/ocs-ci/ocs-ci-13-increase-timeout.patch ../../files/ocs-ci/ocs-ci-14-apply-icsp.patch ../../files/ocs-ci/ocs-ci-15-log-rotate.patch ../../files/ocs-ci/ocs-ci-16-nginx-fio-image.patch
ls: cannot access '../../files/ocs-ci/kvm/ocs-ci-*[0-9][0-9]-*.patch': No such file or directory
platform_patchfiles=
Generating consolidated patch file /root/shilpi-rebase/ocs-ci.patch from /root/shilpi-rebase/ocs-upi-kvm/files/ocs-ci/
checking file ocs_ci/ocs/benchmark_operator.py
checking file ocs_ci/ocs/amq.py
checking file ocs_ci/templates/workloads/amq/benchmark/values.yaml
checking file ocs_ci/templates/workloads/amq/hello-world-consumer.yaml
checking file ocs_ci/templates/workloads/amq/hello-world-producer.yaml
checking file tests/cross_functional/scale/test_scale_osds_fill_75%_reboot_workers.py
checking file tests/functional/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py
checking file tests/functional/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py
checking file tests/functional/z_cluster/cluster_expansion/test_delete_pod.py
checking file tests/functional/z_cluster/cluster_expansion/test_node_expansion.py
checking file tests/functional/z_cluster/cluster_expansion/test_verify_ceph_csidriver_runs_on_non_ocs_nodes.py
checking file tests/functional/z_cluster/cluster_expansion/test_crashcollector_pod_existence_on_ceph_pods_running_nodes.py
checking file tests/functional/z_cluster/nodes/test_node_replacement_proactive.py
checking file tests/cross_functional/scale/test_pv_scale_and_respin_ceph_pods.py
checking file ocs_ci/ocs/node.py
checking file ocs_ci/deployment/deployment.py
checking file ocs_ci/templates/workloads/amq/kafkadrop.yaml
checking file ocs_ci/ocs/resources/storage_cluster.py
checking file ocs_ci/ocs/resources/pod.py
checking file ocs_ci/templates/workloads/logwriter/cephfs.logreader.yaml
checking file ocs_ci/templates/workloads/logwriter/cephfs.logwriter.yaml
checking file ocs_ci/templates/workloads/logwriter/cephfs.reproducer.yaml
checking file ocs_ci/deployment/helpers/lso_helpers.py
checking file tests/functional/z_cluster/test_ceph_default_values_check.py
checking file tests/functional/z_cluster/test_must_gather.py
checking file tests/functional/object/mcg/test_s3_with_java_sdk.py
checking file tests/functional/object/mcg/test_noobaa_secret.py
checking file tests/functional/z_cluster/test_hugepages.py
checking file tests/functional/z_cluster/test_rook_ceph_operator_log_type.py
checking file tests/functional/monitoring/prometheus/alerts/test_noobaa.py
checking file tests/functional/object/mcg/test_mcg_resources_disruptions.py
checking file tests/functional/object/mcg/test_host_node_failure.py
checking file tests/functional/z_cluster/nodes/test_nodes_restart.py
checking file ocs_ci/helpers/helpers.py
checking file ocs_ci/utility/deployment.py
checking file tests/functional/z_cluster/test_rook_ceph_log_rotate.py
checking file ocs_ci/templates/CSI/cephfs/pod.yaml
checking file ocs_ci/templates/CSI/rbd/pod.yaml
checking file ocs_ci/templates/app-pods/nginx.yaml
checking file ocs_ci/templates/app-pods/raw_block_pod.yaml
Patching ocs-ci...
patching file ocs_ci/ocs/benchmark_operator.py
patching file ocs_ci/ocs/amq.py
patching file ocs_ci/templates/workloads/amq/benchmark/values.yaml
patching file ocs_ci/templates/workloads/amq/hello-world-consumer.yaml
patching file ocs_ci/templates/workloads/amq/hello-world-producer.yaml
patching file tests/cross_functional/scale/test_scale_osds_fill_75%_reboot_workers.py
patching file tests/functional/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py
patching file tests/functional/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py
patching file tests/functional/z_cluster/cluster_expansion/test_delete_pod.py
patching file tests/functional/z_cluster/cluster_expansion/test_node_expansion.py
patching file tests/functional/z_cluster/cluster_expansion/test_verify_ceph_csidriver_runs_on_non_ocs_nodes.py
patching file tests/functional/z_cluster/cluster_expansion/test_crashcollector_pod_existence_on_ceph_pods_running_nodes.py
patching file tests/functional/z_cluster/nodes/test_node_replacement_proactive.py
patching file tests/cross_functional/scale/test_pv_scale_and_respin_ceph_pods.py
patching file ocs_ci/ocs/node.py
patching file ocs_ci/deployment/deployment.py
patching file ocs_ci/templates/workloads/amq/kafkadrop.yaml
patching file ocs_ci/ocs/resources/storage_cluster.py
patching file ocs_ci/ocs/resources/pod.py
patching file ocs_ci/templates/workloads/logwriter/cephfs.logreader.yaml
patching file ocs_ci/templates/workloads/logwriter/cephfs.logwriter.yaml
patching file ocs_ci/templates/workloads/logwriter/cephfs.reproducer.yaml
patching file ocs_ci/deployment/helpers/lso_helpers.py
patching file tests/functional/z_cluster/test_ceph_default_values_check.py
patching file tests/functional/z_cluster/test_must_gather.py
patching file tests/functional/object/mcg/test_s3_with_java_sdk.py
patching file tests/functional/object/mcg/test_noobaa_secret.py
patching file tests/functional/z_cluster/test_hugepages.py
patching file tests/functional/z_cluster/test_rook_ceph_operator_log_type.py
patching file tests/functional/monitoring/prometheus/alerts/test_noobaa.py
patching file tests/functional/object/mcg/test_mcg_resources_disruptions.py
patching file tests/functional/object/mcg/test_host_node_failure.py
patching file tests/functional/z_cluster/nodes/test_nodes_restart.py
patching file ocs_ci/helpers/helpers.py
patching file ocs_ci/utility/deployment.py
patching file tests/functional/z_cluster/test_rook_ceph_log_rotate.py
patching file ocs_ci/templates/CSI/cephfs/pod.yaml
patching file ocs_ci/templates/CSI/rbd/pod.yaml
patching file ocs_ci/templates/app-pods/nginx.yaml
patching file ocs_ci/templates/app-pods/raw_block_pod.yaml
Requirement already satisfied: pip in /root/shilpi-rebase/venv/lib/python3.9/site-packages (21.2.3)
Collecting pip
```

### CODE REVIEW

File: scripts/helper/create-cluster.sh [hunk 1]
```diff
@@ -11,110 +11,56 @@ ARG1=$1
...
                 if [ -z "$RHCOS_RELEASE" ]; then
                         RHCOS_RELEASE="4.15.0"                   # Latest release of RHCOS 4.14 at this time
                 fi
                 RHCOS_SUFFIX="-$RHCOS_RELEASE"
                 ;;	
-	4.16)
-                unset OCP_RELEASE
+        4.16)
+                OCP_RELEASE="4.16.3"
                 RHCOS_VERSION="4.15"
```
Comment (AaruniAggarwal) [inline scripts/helper/create-cluster.sh:49]: It should be RHCOS_VERSION="4.16"

File: scripts/helper/create-cluster.sh [hunk 2]
```diff
@@ -11,110 +11,56 @@ ARG1=$1
...
+                if [ -z "$RHCOS_RELEASE" ]; then
+                        RHCOS_RELEASE="4.16.0"                   # Latest release of RHCOS 4.14 at this time
+                fi
+                RHCOS_SUFFIX="-$RHCOS_RELEASE"
+                ;;
+
+	4.17)
+                unset OCP_RELEASE
+                RHCOS_VERSION="4.17"
```
Comment (AaruniAggarwal) [inline scripts/helper/create-cluster.sh:58]: Here RHCOS_VERSION should be 4.16
@Shilpi-Das1

File: scripts/run-fio.sh [hunk 3]
```diff
@@ -237,7 +237,7 @@ oc project default >/dev/null 2>&1
 ocs_operator=`oc get csv -n openshift-storage | grep ocs-operator | awk {'print $1'}`
 ocs_version=`oc get csv $ocs_operator -n openshift-storage -o jsonpath={.spec.version} | cut -d "." -f 1-2`
 cephblockpool=$(oc -n openshift-storage rsh $ceph_tools ceph df | grep ocs-storagecluster-cephblockpool)
-if [[ $ocs_version == "4.9" || $ocs_version == "4.10" || $ocs_version == "4.11" || $ocs_version == "4.12" || $ocs_version == "4.13" || $ocs_version == "4.14" || $ocs_version == "4.15" || $ocs_version == "4.16" ]]; then
+if [[ $ocs_version == "4.12" || $ocs_version == "4.13" || $ocs_version == "4.14" || $ocs_version == "4.15" || $ocs_version == "4.16" || $ocs_version == "4.17" ]]; then
```
Comment (AaruniAggarwal) [inline scripts/run-fio.sh:240]: As Supported ODF versions are now from 4.12 onwards, we can remove if-else condition. 
WDYT, @Shilpi-Das1 ?

File: scripts/helper/create-cluster.sh (hunk 2 above)
Comment (Shilpi-Das1) [reply scripts/helper/create-cluster.sh]: Changed it

File: scripts/run-fio.sh (hunk 3 above)
Comment (Shilpi-Das1) [reply scripts/run-fio.sh]: Agree
Removed if else

**Comment by AaruniAggarwal** [comment]
Have checked the PR and executed the setup script from Shilpi's PR and patchfiles are applying successfully.
/LGTM

**Comment by AaruniAggarwal** [comment]
/LGTM 

Merging the PR.
Thanks @Shilpi-Das1

**Pull request merged**